packages/viem/src/index.ts

```diff
@@ -10,11 +10,69 @@ import type {
   TransactionSerializable,
   TypedData,
 } from "viem";
-import { TurnkeyActivityError, TurnkeyClient } from "@turnkey/http";
+import {
+  assertNonNull,
+  checkActivityStatus,
+  TActivityStatus,
+  TActivityId,
+  TSignature,
+  TurnkeyActivityError as TurnkeyHttpActivityError,
+  TurnkeyActivityConsensusNeededError as TurnkeyHttpActivityConsensusNeededError,
+  TurnkeyClient,
+} from "@turnkey/http";
 import { ApiKeyStamper } from "@turnkey/api-key-stamper";
 import type { TurnkeyBrowserClient } from "@turnkey/sdk-browser";
 import type { TurnkeyServerClient } from "@turnkey/sdk-server";
 
+export type TTurnkeyConsensusNeededErrorType = TurnkeyConsensusNeededError & {
+  name: "TurnkeyConsensusNeededError";
+};
+export class TurnkeyConsensusNeededError extends BaseError {
```

Comment:
Instead of defining these errors here, can we pull that into `http`? Is there a way to comply with viem's error interface (implement `walk` is one thing; maybe others?) without importing viem's BaseError directly?

We'd need to implement this interface in our http/shared errors: https://github.com/wevm/viem/blob/408948bb74685e9d562cd39f50768f9b282463fe/src/errors/base.ts#L37-L83

I don't think it'd hurt anything to have our errors implement more methods than they need to, but maybe not something we need right now; feels a bit awkward to be pushed by a particular framework to implement these methods for all Turnkey errors.

Reply:
+1 on the sentiment. I'm down to keep these encapsulated to just Viem for now